src/math/fmod.c

```diff
@@ -0,0 +1,22 @@
+/* This is free and unencumbered software released into the public domain. */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+/**
+ * @date   2017-10-02
+ * @author Samuel Sarle
+ * @see    http://libc11.org/math/fmod.html
+ */
+
+double
+fmod(double x, const double y) {
+  if (x > y && x > 0 && y > 0) {
+    while (x >= y) {
+      x -= y;
+    }
+    return x;
+  }
+  return 0;
+}
```

Comment:
Going forward, reverse this outermost `if` such that the exceptional control path (`return 0`) is handled first, and the function otherwise proceeds to the normal control path (the loop) without an extra level of indentation.